### ISSUE TITLE
Improve the timeout algorithm

### ISSUE THREAD

**Issue by ming871** [lsalzman/enet#40]
Enet used timeout algorithm is too simple, when use it on the Internet, disconnection often happen.

**Comment by jiyonglong** [comment]
yes,  I also encountered the same problem, and i don't know how to solve it.  do somebody have some suggestions ? thank you !

**Comment by jiyonglong** [comment]
how to solve the problem ? nobody can ?

**Comment by ElementW** [comment]
What better algorithm do you suggest? Currently, the algorithm is a basic timeout one: a peer's timeout time point is reset on every interaction (anything received from the client), and timeout check and handling is done in [`enet_protocol_check_timeouts()`](https://github.com/lsalzman/enet/blob/f46fee0acc8e243b2b6910b09693f93c3aad775f/protocol.c#L1414).

AFAIK timeout is a simple thing to handle and there is no smart(er) way of doing it.
